LibOS/shim/src/syscallas.S

```diff
@@ -109,3 +109,41 @@ ret:
 
         .cfi_endproc
         .size syscalldb, .-syscalldb
+
+
+        /*
+         * syscall_wrapper served as a redirection point
+         * for static system calls captured through
+         * host-lvel exception handling. Its jobs are:
+         * (1) reserving red zone (128 bytes)
+         * (2) call syscalldb
+         * (3) return to the original code (saved in RBX)
+         *
+         * Chia-Che 12/13/17:
+         * There is a risk that this code may corrupt RBX.
+         * I haven't figure out how to preserve RBX because
+         * the code needs to sacrifice a register to do
+         * the final jump.
```

Comment:
What is the problem exactly? Sounds like a thing which could be solved without too much effort.
Is the problem about this function taking the argument in rbx or about overwriting rbx which was set somewhere inside `callq *%rbx` ?